tensorflow/compiler/mlir/hlo/BUILD

```diff
@@ -717,6 +717,35 @@ cc_library(
     alwayslink = 1,
 )
 
+cc_library(
+    name = "lower_alloc_dealloc",
```

Comment:
This library isn't used?

Reply:
It cannot be used right now. This PR depends on PR "legalize disc to llvm".  I will move these two patterns to ral_legalize_to_llvm.cc once "legalize disc to llvm" is merged.

Reply:
mv to tensorflow/compiler/mlir/hlo/lib/Dialect/mhlo/transforms/ral_legalize_to_llvm.cc